Cmdline/Action/List.cs

```diff
@@ -65,26 +65,49 @@ public int RunCommand(CKAN.KSP ksp, object raw_options)
                     else if (current_version is DllVersion)
                     {
                         // Autodetected dll
-                        bullet = "-";
+                        bullet = "A";
                     }
                     else
                     {
                         try
                         {
                             // Check if upgrades are available, and show appropriately.
                             CkanModule latest = registry.LatestAvailable(mod.Key, ksp.VersionCriteria());
-
+                            CkanModule current = registry.GetInstalledVersion(mod.Key);
+                                
                             log.InfoFormat("Latest {0} is {1}", mod.Key, latest);
 
                             if (latest == null)
                             {
                                 // Not compatible!
                                 bullet = "X";
+                                //Check if mod is replaceable
+                                if ( current.replaced_by != null)
+                                {
+                                    CkanModule replacement = registry.LatestAvailable(current.replaced_by.name, ksp.VersionCriteria());
+                                    if (replacement != null)
+                                    {
+                                        // Replaceable
+                                        bullet = ">";
+                                        modInfo = string.Format("{0} {1} > {2} {3}", mod.Key, mod.Value, replacement.name, replacement.version);
+                                    }
+                                }
                             }
                             else if (latest.version.IsEqualTo(current_version))
                             {
                                 // Up to date
                                 bullet = "-";
+                                //Check if mod is replaceable
```

Comment:
As a general rule, code that's needed in multiple places should be refactored to a function and called from both places rather than copied. Can you do that with these blocks?

Reply:
Yep, and I can then do more detailed testing on the replaced_by version criteria

Reply:
I like the symmetry between `modInfo` and `newModInfo` now.